### ISSUE TITLE
Adopt either command_center or legendary

### ISSUE THREAD

**Issue by danielo515** [LunarVim/LunarVim#2671]
### Feature Description

I want to open a discussion about integrating [command_center](https://github.com/FeiyouG/command_center.nvim) or [legendary](https://github.com/mrjones2014/legendary.nvim) into lunarvim.
Those are projects that try to bring the rich editor experience of the VSCode omnibar into neovim.
The benefits that I envision are many:

- Better discoverability of the existing commands and keyboard shortcuts. Sure which-key does a good job there, but there is certain level of nesting which after that it is too buried to be found. The show logs are a great example `LlD`, they are three levels deep behind two capital letters. If you have to check them often it is uncomfortable and you may use lunarvim for weeks without noticing they are there. On the other hand, opening an omnibar and typing what you think should be there (here, `log`) you will find all the possible show log options. I often use this in VSCode to re-discover features that I don't use often. I just open the command palette and type what I remember from the command name in plain english and I often find that thing that I use once a month.
- Better discoverability of existing key maps. Just like in the first point, this tools not only give you a great explanation of the command, they also offer you a remind of the actual keybinding if there is any.
- More options are always better 😄 . Having the option to use the keyboard directly or go the command palette route is always something nice to have.

For extra familiarity for users coming from VSCode we can bind this to `ctrl + P` and they will find right at home!

I'll highlight the differences between the two proposed projects in another comment later.

### Describe the alternatives you have considered

I have this configured in my local config

**Comment by kobzar** [comment]
For my opinion - its no VIM like style.
Vimer must understand what they do.
Thats why we are edit cinfig and do map/remap keys.
For the quick help SPS + L + k  is anough
But its for me only. Ofcource you can use any external plugins, that U like, but its bad idea to make from
quick editor the same monster as vscode

**Comment by danielo515** [comment]
> For my opinion - its no VIM like style.

Luckily we are not talking about VIM here! Not even NeoVIM! This is LunarVim! 😄 
Jokes aside, I think there is no VIM way, because it is just a tool, an editor, and everybody has their opinions and likings regarding it.

I tried to use nvim as my main IDE for several years now, failing miserably on each attempt. It was not until more friendly tools like telescope, lsp helpers, popups, nvim-tree, lua-line etc came up that I was able to use it effectively, and I was not until I started using LVIM that I found a configuration good enough to consider neovim a good IDE. Does that qualify me as a worse nvim user? I don't think so.

Tools like this really help me be efficient without having to remember key combinations that I don't use often. Will a key combination that is specific to one buffer type and that you use once in a month get into your muscle memmory? I don't think so. Do you want to have to check your keymaps every-time you want to use it? I don't think so either. I just spin my "omnibar" type what the command is about and I get it with a reminder of the keymap in case I want to learn it. And some commands are so uncommon that they don't even deserve a keybboard shortcut.

IMO giving the users that want them a more friendly and visual interface, while keeping all the existing keyboard shortcuts for more power users it is a win-win situation where I don't see the problem in having.

**Comment by rebuilt** [comment]
Keybindings are very personal.  The fewer bindings we include, the better.  There has been a lot of churn over what bindings we put in.  And at the end of the day keybindings come down to personal taste.  

I'm not convinced about the better discoverability by adding an infrastructure to search for bindings.  Removing bindings will just as easily make things more discoverable.  

Also, I'm not clear on the specifics of this suggestions.  Does this mean we rip out whichkey to add one of these plugins?  Can the keybindings LunarVim defines with command_center then be overwritten by the end user?

**Comment by danielo515** [comment]
> Keybindings are very personal. The fewer bindings we include, the better. There has been a lot of churn over what bindings we put in. And at the end of the day keybindings come down to personal taste.

Couldn't agree more. However, most editors loved by millions still include some keybindings, however, most of them also provide a command palette so you don't need to learn them all nor configure them all. I used VSCode for years and I only learn a few key bindings because as soon as I needed anything beyond my daily usage I could just open the command palette, type what I thought I wanted (check the keybinding in case I were interested) and continue working after executing the command.

> I'm not convinced about the better discoverability by adding an infrastructure to search for bindings. Removing bindings will just as easily make things more discoverable.

The issue is not just about key-bindings, it is about having a more human friendly interface than having to remember either keybindings or commands. I can't express how much nicer is to have a search interface when you are not sure of how the thing you want to do is configured (keybinding) or called (in case of commands). Which-key solves this to some degree, but you still need to navigate all the sections if you want to do where each thing is, and still you have to keep in your mind in which section certain command is.

> Also, I'm not clear on the specifics of this suggestions. Does this mean we rip out whichkey to add one of these plugins? Can the keybindings LunarVim defines with command_center then be overwritten by the end user?

Nope, I'm suggesting to include one of this plugins additionally to which-key. I use them both and each covers an specific use-case in my daily work. When I'm clear about what I want to do but I don't remember the last key of a key combination (was `<space>lf` or `<space>ls` for symbols?). When I'm not sure if what I want to do is a key-binding, or a command or the name of the command I then open my command-palette and then I type in the description or the command fuzzily.

I think we can include all LunarVim commands with proper descriptions, and most used vim built-in commands with proper descriptions easily for a nicer overall experience. In fact, legendary already includes several vim keymaps and commands with proper descriptions built-in along with several keymaps with proper descriptions and allows you to search-then-execute both in the same place.

Hope the proposition makes more sense now.

**Comment by luxus** [comment]
from my understanding, the whichkey shortcuts are integrated. so most, if not all keybindings of lunar vim would be visible with no manual work except adding the plugin.
am I right?

**Comment by kobzar** [comment]
> from my understanding, the whichkey shortcuts are integrated. so most, if not all keybindings of lunar vim would be visible with no manual work except adding the plugin.
> am I right?

No. He talk about command palete, like vscide,st2 e.t.c
You join in command line like vim cmd, but can make any action in editor that you can pin to keys. Just write file format and this format the file

**Comment by danielo515** [comment]
> from my understanding, the whichkey shortcuts are integrated. so most, if not all keybindings of lunar vim would be visible with no manual work except adding the plugin. am I right?

If we opt for legendary, then yes. We will need to pass the which key config to the plugin setup, but that it's all that is required.

@kobzar if we opt to use Legendary instead of command palette then @luxus is right.

**Comment by luxus** [comment]
from my understanding @abzcoding supports command_center and its not very helpful. legendary could be better.. but it shouldn't be in the core. every plugin will add more overhead to the maintainers. its possible to add it by urself or ask for help how to do it and add the infos to the wiki. there is no reason to make it a "core" plugin
